x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/AsyncStatusResponse.java

```diff
@@ -132,9 +150,13 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeVInt(successfulShards);
         out.writeVInt(skippedShards);
         out.writeVInt(failedShards);
-        if (isRunning == false) {
+        if (isRunning == false) { /// MP: Hmm, is this going to be a problem? Why is this not writeOptionalString?
             RestStatus.writeTo(out, completionStatus);
         }
+        if (out.getTransportVersion().onOrAfter(TransportVersion.V_8_500_009)) {
+            // optional since only CCS uses is; it is null for local-only searches
```

Comment:
This needs to be nullable because older nodes won't send it over the wire. I am less clear on having it null for local-only searches. Shouldn't we use EMPTY in this case like we normally do?

Reply:
Currently (before this ticket) `_clusters` is not shown for local-only searches in the async-status/status response, so that has been the default already.

And in addition, I'm not sure why we use EMPTY for the normal non-CCS case, as that seems wrong to me. 

This type of output would be confusing to end users:
```
{
  "id": "FnhKR2FES241UmNtSjhGWUVTYkhGWWcaemJJaENTNW9SZFdaWnNmRUNzbE1VUTozNjU",
  "is_running": false,
  "is_partial": false,
  "start_time_in_millis": 1686227754043,
  "expiration_time_in_millis": 1686659754043,
  "_shards": {
    "total": 3,
    "successful": 3,
    "skipped": 0,
    "failed": 0
  },
  "_clusters": {
    "total": 0,  // why is this zero?
    "successful": 0, // didn't my local cluster search work?
    "skipped": 0
  },
  "completion_status": 200
}
```

IMO, the standard non-CCS default Clusters object should be `Clusters{total=1, successful=1, skipped=0}`. Using empty implies that we don't want to count the local cluster as a cluster. But if that's true, then should we also not count it for CCS? But then that would be also confusing to end users.

If we add EMPTY here for the local-only, we could choose to not serialize it, but that requires extra logic in the code above or if we serialize it but don't want to show it, then that requires extra logic in the toXContent methods. So I chose the path of least resistance which is to just have it be `null`, which matches current functionality for local-only searches.

If you want EMPTY serialized, then I would say we need to add extract logic in `toXContent` to not output it to the REST response.